arangodb-foxx-services/src/main/services/prune-database.ts

```diff
@@ -0,0 +1,130 @@
+            FOR progOf IN progressOf
+                FILTER execPlan._created < ${timestamp} && progOf._to == execPlan._id
+                COLLECT ids = execPlan._id
+                RETURN ids
+    `).toArray()
+
+    const orphanExecPlanIDsOlderThanThresholdDaysArray = db._query(aql`
+        FOR execPlan IN executionPlan
+            FILTER execPlan._id NOT IN ${execPlanIDsWithProgressEventsArray} && execPlan._created < ${timestamp}
+            RETURN execPlan._id
+    `).toArray()
+
+    const collections = ['executes', 'operation', 'follows', 'uses', 'expression', 'affects', 'depends', 'writesTo', 'readsFrom', 'emits', 'produces', 'consistsOf', 'derivesFrom', 'computedBy', 'takes', 'schema', 'attribute']
+    for (let i = 0; i < collections.length; i++) {
+        console.log('### Working on', collections[i], 'collection')
+        const startCount = db._query('RETURN COUNT(@@cols)', {'@cols': collections[i]}).toArray()[0]
+        db._query('FOR orphanExecPlanID IN @arr FOR collectionEle IN @@cols FILTER collectionEle._belongsTo == orphanExecPlanID && collectionEle._created < @purgeLimitTimestamp REMOVE collectionEle IN @@cols',
+            {
+                'arr': orphanExecPlanIDsOlderThanThresholdDaysArray,
+                '@cols': collections[i],
+                'purgeLimitTimestamp': timestamp
+            }
+        )
+        const endCount = db._query('RETURN COUNT(@@cols)', {'@cols': collections[i]}).toArray()[0]
+        console.log(startCount - endCount, collections[i], 'objects deleted...')
+    }
+
+    ////////////////////////////////////////////////////////////////////////
+    // reused below
+    let startCount = 0
+    let endCount = 0
+
+    console.log('### Working on executionPlan collection')
+    startCount = db._query('RETURN COUNT(@@cols)', {'@cols': 'executionPlan'}).toArray()[0]
+    db._query(aql`
+        FOR doc IN executionPlan
+            FILTER doc._id IN ${orphanExecPlanIDsOlderThanThresholdDaysArray}
+            REMOVE doc IN executionPlan
+    `)
+    endCount = db._query('RETURN COUNT(@@cols)', {'@cols': 'executionPlan'}).toArray()[0]
+    console.log(startCount - endCount, 'executionPlan objects deleted...')
+
+    // Deleted other collections and then the orphanExecPlanIDs
+
+    const t2 = Date.now()
+    console.log(`All collections but dataSource purge in ${t2 - t1} ms`)
+
+    ////////////////////////////////////////////////////////////////////////
+
+
+    // STAGE 3: No `affects` and `depends` corresponding to dataSource (Need to scan all dataSources, and both edges foreach) -> delete dataSource
+    const dataSourceKeysInAffectUseArray = db._query(`
+        FOR source IN dataSource
+            FOR affectEdge in affects
+                FILTER affectEdge._to == source._id
+                COLLECT keys = source._key
+                RETURN keys
+    `).toArray()
+
+    const dataSourceKeysInAffectsDependsUseArray = dataSourceKeysInAffectUseArray.concat(
+        db._query(`
```

Comment:
Minor: using a back tick for string literal, should be just `'`

Reply:
In order to avoid squashing everything to one line or concatenating multiple lines of literal strings, backticks or `aql`backticks seem appropriate -- looking on the doc https://www.arangodb.com/docs/stable/foxx-guides-queries.html

Reply:
Ah, my bad, sorry. I somehow overlooked it was a multiline AQL query.